File: education_app/lib/features/home/domain/entities/product.dart
import 'package:equatable/equatable.dart';

class Product extends Equatable {
  final int id;
  final String title;
  final String description;
  final String category;
  final String imageUrl;
  final double price;

  const Product({
    required this.id,
    required this.title,
    required this.description,
    required this.category,
    required this.imageUrl,
    required this.price,
  });

  @override
  List<Object?> get props => [id, title, description, category, imageUrl, price];
}

File: Backend/DigitalStore.Domain/Entities/Product.cs
using DigitalStore.Domain.Common;

namespace DigitalStore.Domain.Entities
{
    public class Product : BaseEntity
    {
        public required string Title { get; set; }
        public string? Description { get; set; }
        public string? Category { get; set; } // e.g., "Grade 6", "Math"
        public string? ImageUrl { get; set; }
        
        [System.ComponentModel.DataAnnotations.Schema.Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
    }
}
